llvm/lib/Target/X86/X86ISelDAGToDAG.cpp

```diff
@@ -5294,11 +5294,26 @@ void X86DAGToDAGISel::Select(SDNode *Node) {
       return;
     if (tryVPTERNLOG(Node))
       return;
-
     [[fallthrough]];
   case ISD::ADD:
     if (Opcode == ISD::ADD && matchBitExtract(Node))
       return;
+
+    // Convert addlike to add before final selection. Do this before we drop
+    // flags like `disjoint`.
```

Comment:
Where does the flag get dropped?

Reply:
in SelectCode i assume, when we get to `or_is_add` didn't have `disjoint` or `ISD::OR`.

Reply:
You're right, I re-checked. I must have made a mistake earlier.
It doesn't seem to work though (I thought I had the reason, but now I'm stumped).
If I change it to your code 104 of the changes files revert back.
Why would that be the case?

Reply:
We have special pseudo instruction  we use for or_is_add. ADD_DB. It's convertible to LEA but it always converts back to OR if no LEA is needed.

Reply:
ahh, so this patch is essentially not useful, it should just be what you have.

Reply:
Ill update, thank you for explaining.